### PR TITLE
podman-remote make --size optional in ps

### LINKED ISSUE #3578: `podman-remote ps` has `--size` set by default
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

By default `podman-remote ps` computes all containers' rootfs and rw size, and this results in significant higher response time compared to running `podman ps`.

Effect on a system with 5 containers:
```
$ time sudo ./bin/podman-remote ps
CONTAINER ID  IMAGE  COMMAND  CREATED  STATUS  PORTS  NAMES
```
/usr/bin/sudo -E ./bin/podman-remote ps  0.03s user 0.01s system 0% cpu **24.108 total**
```
$ time sudo ./bin/podman ps
CONTAINER ID  IMAGE  COMMAND  CREATED  STATUS  PORTS  NAMES
```
/usr/bin/sudo -E ./bin/podman ps  0.04s user 0.03s system 100% cpu **0.075 total**

### PR DESCRIPTION
**Pull request by QiWang19**
Close #3578 Add `size` field to PsOpts in podman remote to calculate container size when necessary.

Signed-off-by: Qi Wang <qiwan@redhat.com>

### CODE REVIEW

File: cmd/podman/varlink/io.podman.varlink [hunk 1]
```diff
@@ -144,10 +144,11 @@ type PsOpts (
...
     noTrunc: ?bool,
-	pod: ?bool,
-	quiet: ?bool,
-	sort: ?string,
-	sync: ?bool
+    pod: ?bool,
+    quiet: ?bool,
+    sort: ?string,
+    sync: ?bool,
+    size: ?bool
```
Comment (mheon) [inline cmd/podman/varlink/io.podman.varlink]: Can we order these alphabetically? Size before sort and sync?

Comment (QiWang19) [reply cmd/podman/varlink/io.podman.varlink]: fixed

**Comment by TomSweeneyRedHat** [comment]
LGTM assuming happy tests

File: pkg/varlinkapi/util.go [hunk 2]
```diff
@@ -191,7 +191,7 @@ func makePsOpts(inOpts iopodman.PsOpts) shared.PsOptions {
 		Latest:    derefBool(inOpts.Latest),
 		NoTrunc:   derefBool(inOpts.NoTrunc),
 		Pod:       derefBool(inOpts.Pod),
-		Size:      true,
+		Size:      derefBool(inOpts.Sync),
```
Comment (giuseppe) [inline pkg/varlinkapi/util.go]: shouldn't this be `derefBool(inOpts.Sort)`?

Comment (QiWang19) [reply pkg/varlinkapi/util.go]: fixed. Should be derefBool(inOpts.Size)

Comment (TomSweeneyRedHat) [reply pkg/varlinkapi/util.go]: nice catch @giuseppe, I missed that.

**Comment by TomSweeneyRedHat** [comment]
LGTM with happy tests.

**Comment by mheon** [comment]
LGTM once things go green
/lgtm

**Comment by TomSweeneyRedHat** [comment]
Does @giuseppe 's hold have to be canceled for this to go through?

**Comment by mheon** [comment]
Yeah. We use the hold to make sure it doesn't land before tests go green.
/hold cancel

**Pull request merged**